src/analysis/inner_product_space/positive.lean

```diff
@@ -124,3 +126,198 @@ end
+    rw [linear_map.sub_apply, inner_sub_left, ← inner_conj_sym, linear_map.adjoint_inner_left,
+        ← right, is_R_or_C.conj_of_real, sub_self], },
+   refine ⟨frs,_⟩,
+   intros μ hμ,
+   rw ← module.End.has_eigenvalue_iff_mem_spectrum at hμ,
+   have realeigen := (complex.eq_conj_iff_re.mp
+     (linear_map.is_symmetric.conj_eigenvalue_eq_self
+       ((linear_map.is_symmetric_iff_is_self_adjoint T).mpr frs) hμ)).symm,
+   refine ⟨realeigen, _⟩,
+   have hα : ∃ α : ℝ, α = μ.re := by use μ.re,
+   cases hα with α hα,
+   rw ← hα,
+   rw [realeigen, ← hα] at hμ,
+   exact eigenvalue_nonneg_of_nonneg hμ (λ x, ge_iff_le.mp (h x).1),
+end
+
+variables [finite_dimensional ℂ V] {n : ℕ} (hn : finite_dimensional.finrank ℂ V = n)
+variables (T : V →ₗ[ℂ] V)
+
+lemma of_pos_eq_sqrt_sqrt (hT : T.is_symmetric)
+  (hT1 : ∀ μ : ℂ, μ ∈ spectrum ℂ T → μ = ↑μ.re ∧ 0 ≤ μ.re) (v : V) :
+  T v = ∑ (i : (fin n)), real.sqrt (hT.eigenvalues hn i) • real.sqrt (hT.eigenvalues hn i)
+   • ⟪(e hT hn) i, v⟫_ℂ • (e hT hn) i :=
+begin
+  have : ∀ i : fin n, 0 ≤ hT.eigenvalues hn i := λ i,
+  by { specialize hT1 (hT.eigenvalues hn i),
+      simp only [complex.of_real_re, eq_self_iff_true, true_and] at hT1,
+      apply hT1 (module.End.mem_spectrum_of_has_eigenvalue
+        (is_symmetric.has_eigenvalue_eigenvalues hT hn i)), },
+  calc T v = ∑ (i : (fin n)), ⟪(e hT hn) i, v⟫_ℂ • T ((e hT hn) i) :
+  by simp_rw [← orthonormal_basis.repr_apply_apply, ← map_smul_of_tower, ← linear_map.map_sum,
+                orthonormal_basis.sum_repr (is_symmetric.eigenvector_basis hT hn) v]
+       ... = ∑ (i : (fin n)),
+        real.sqrt (hT.eigenvalues hn i) • real.sqrt (hT.eigenvalues hn i) •
+         ⟪(e hT hn) i, v⟫_ℂ • (e hT hn) i :
+  by simp_rw [is_symmetric.apply_eigenvector_basis, smul_smul, ← real.sqrt_mul (this _),
+              real.sqrt_mul_self (this _), mul_comm, ← smul_smul, complex.coe_smul],
+end
+
+include hn
+/-- Let `e = hT.eigenvector_basis hn` so that we have `T (e i) = α i • e i` for each `i`.
+Then when `T.is_symmetric` and all its eigenvalues are nonnegative,
+we can define `T.sqrt` by `e i ↦ √α i • e i`. -/
+noncomputable def sqrt (hT : T.is_symmetric) : V →ₗ[ℂ] V :=
+{ to_fun := λ v, ∑ (i : (fin n)),
+             real.sqrt (hT.eigenvalues hn i) • ⟪(hT.eigenvector_basis hn) i, v⟫_ℂ
+              • (hT.eigenvector_basis hn) i,
+  map_add' := λ x y, by simp_rw [inner_add_right, add_smul, smul_add, finset.sum_add_distrib],
+  map_smul' := λ r x, by simp_rw [inner_smul_right, ← smul_smul, finset.smul_sum,
+                                  ring_hom.id_apply, ← complex.coe_smul, smul_smul,
+                                  ← mul_assoc, mul_comm] }
+
+lemma sqrt_eq (hT : T.is_symmetric) (v : V) : (T.sqrt hn hT) v = ∑ (i : (fin n)),
+  real.sqrt (hT.eigenvalues hn i) • ⟪(hT.eigenvector_basis hn) i, v⟫_ℂ
+   • (hT.eigenvector_basis hn) i := rfl
+
+/-- `T.sqrt ^ 2 = T` and `T.sqrt.is_positive` -/
+lemma sqrt_sq_eq_linear_map_and_is_positive (hT : T.is_symmetric)
```

Comment:
This lemma proves two things, so it should be two lemmas. Further I think both should be tagged `@[simp]`.